From b7001315fc2fbfc82f008d5e2425b1ff88c33d6e Mon Sep 17 00:00:00 2001
From: Samuel Ortiz <sameo@linux.intel.com>
Date: Mon, 8 Sep 2014 14:17:32 +0200
Subject: [PATCH 1/2] iio: bmm050: Soft reset on the resume path

When resuming some register fails to be written when resuming, leading
to a 3 seconds delay on the system resume path.
Soft resetting the chipset on the resume path fixes that problem.

Issue: IRDA-710
Change-Id: I9428fe5054327238854412d69add5e3aa42b0d47
Signed-off-by: Samuel Ortiz <sameo@linux.intel.com>
---
 drivers/iio/magnetometer/bmm050_driver.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/iio/magnetometer/bmm050_driver.c b/drivers/iio/magnetometer/bmm050_driver.c
index 8719c85..7572613 100644
--- a/drivers/iio/magnetometer/bmm050_driver.c
+++ b/drivers/iio/magnetometer/bmm050_driver.c
@@ -1466,6 +1466,14 @@ static int bmm_resume(struct device *dev)
 
 	dev_dbg(&client->dev, "function entrance %s", __func__);
 
+	/* Soft reset the chipset */
+	BMM_CALL_API(soft_reset)();
+	mdelay(BMM_I2C_WRITE_DELAY_TIME);
+
+	/* Wake it up */
+	bmm_wakeup(client);
+
+	/* And then restore the HW configuration */
 	mutex_lock(&client_data->mutex_power_mode);
 	err = bmm_restore_hw_cfg(client);
 	/* post resume operation */
-- 
1.8.3.2

